src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java

```diff
@@ -911,6 +912,13 @@ else if((forceReplace || newRecipe.getMustExist()) && duplicateFound != recipeTo
      */
     public String getJobDisplayName()
     {
+        if(jobDisplayName.isEmpty())
+        {
+            //This will only happen if getJobDisplayName is called before the first citizen is assigned. 
+            ICitizenData citizen = colony.getCitizenManager().createAndRegisterCivilianData();
+            jobDisplayName = this.createJob(citizen).getName();
+            colony.getCitizenManager().removeCivilian(citizen);
```

Comment:
I still don't like this since it's not side-effect free. I rather have it return empty for now, until we got a propper fix. (It's only a small UI error after all).

Reply:
removing for now